host_provider/providers/azure.py

```diff
@@ -0,0 +1,342 @@
+                 % (self.credential.subscription_id, self.credential.resource_group, name, api_version)
+        nic = self._parse_nic(name, vnet, subnet)
+
+        payload = json.dumps(nic)
+        header = {}
+        self.get_azure_connection()
+        self.azClient.connect(base_url=base_url)
+        self.azClient.add_default_headers(header)
+        self.azClient.connection.request("PUT", action, body=payload, headers=header)
+        resp = self.azClient.connection.getresponse()
+
+        return resp
+
+    def get_network(self, api_version='2020-07-01'):
+        subnet = self.credential.get_next_zone_from(self.credential.subnets)
+        vnet = self.credential.subnets.get(subnet)['name']
+        base_url = self.credential.endpoint
+
+        action = "subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s?api-version=%s" \
+                 % (self.credential.subscription_id, self.credential.resource_group, vnet, api_version)
+
+        header = {}
+        self.get_azure_connection()
+        self.azClient.connect(base_url=base_url)
+        self.azClient.add_default_headers(header)
+        self.azClient.connection.request("GET", action, headers=header)
+        resp = self.azClient.connection.getresponse()
+
+        if resp.status_code == 200:
+            return resp.json()
+
+        else:
+            raise Exception(
+                "Network {} not found".format(vnet)
+            )
+
+    def generate_tags(self, team_name, infra_name, database_name):
+        tags = TeamClient.make_tags(team_name, self.engine)
+        if HOST_ORIGIN_TAG:
+            tags['origin'] = HOST_ORIGIN_TAG
+        tags.update({
+            'engine': self.engine_name,
+            'infra_name': infra_name,
+            'database_name': database_name or ''
+        })
+        return tags
+
+    def create_host_object(self, provider, payload, env, created_host_metadata, static_ip_id, **kw):
+        nic = created_host_metadata.get('nic', '')
+        vm = created_host_metadata.get('vm', '')
+
+        configs = [conf for conf in nic['properties']['ipConfigurations']]
+        primaryIPAddress = [i['properties']['privateIPAddress'] for i in configs][0]
+        identifier = vm['properties']['vmId']
+
+        address = primaryIPAddress
+        host = Host(
+            name=payload['name'], group=payload['group'],
+            engine=payload['engine'], environment=env, cpu=payload['cpu'],
+            memory=payload['memory'], provider=provider.credential.provider,
+            identifier=identifier, address=address,
+            zone=provider.credential._zone
+        )
+        host.save()
+
+        return host
+
+    def _list_vm(self, name, only_status=True, api_version='2020-12-01'):
+        status_only = 'false' if not only_status else 'true'
+        base_url = self.credential.endpoint
+
+        action = "subscriptions/%s/providers/Microsoft.Compute/virtualMachines/?api-version=%s&statusOnly=%s" \
```

Comment:
- Hardcoded url.
- use format instead of %
- do no use \ to break lines, use parentesis